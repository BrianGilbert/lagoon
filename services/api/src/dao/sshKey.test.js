const { Sql } = require('./sshKey');

describe('Sql', () => {
  describe('allowedToModify', () => {
    it('if not admin, should create statement with IN clauses', () => {
      const cred = {
        role: 'user',
        permissions: {
          customers: ['1', '2'],
          projects: [],
        },
      };
      const id = 1;
      const ret = Sql.allowedToModify(cred, id);

      expect(ret).toMatchSnapshot();
    });

    it('if admin, should create statement without IN clause', () => {
      const cred = {
        role: 'admin',
        permissions: {
          customers: [],
          projects: [],
        },
      };
      const id = 1;
      const ret = Sql.allowedToModify(cred, id);

      expect(ret).toMatchSnapshot();
    });
  });

  describe('updateSshKey', () => {
    it('should create a proper query', () => {
      const cred = {};
      const input = {
        id: 1,
        patch: {
          keyType: 'ssh-rsa',
        },
      };
      const ret = Sql.updateSshKey(cred, input);
      expect(ret).toMatchSnapshot();
    });
  });
  describe('selectSshKey', () => {
    it('should create a proper query', () => {
      const ret = Sql.selectSshKey(1);
      expect(ret).toMatchSnapshot();
    });
  });
  describe('selectUnassignedSshKeys', () => {
    it('should create a proper query', () => {
      const ret = Sql.selectUnassignedSshKeys();
      expect(ret).toMatchSnapshot();
    });
  });

  describe('selectAllSshKeys', () => {
    it('if not admin, should create statement with IN clauses', () => {
      const cred = {
        role: 'user',
        permissions: {
          customers: ['1', '2'],
          projects: [],
        },
      };
      const id = 1;
      const ret = Sql.selectAllSshKeys(cred, id);

      expect(ret).toMatchSnapshot();
    });

    it('if admin, should create statement without IN clause', () => {
      const cred = {
        role: 'admin',
        permissions: {
          customers: [],
          projects: [],
        },
      };
      const id = 1;
      const ret = Sql.selectAllSshKeys(cred, id);

      expect(ret).toMatchSnapshot();
    });
  });


});
