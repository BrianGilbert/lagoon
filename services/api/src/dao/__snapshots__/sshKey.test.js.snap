// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql allowedToModify if admin, should create statement without IN clause 1`] = `"select \`sk\`.\`id\`, \`ps\`.\`pid\`, \`cs\`.\`cid\`, IF(COUNT(sk.id) > 0, 1, 0) as allowed from \`ssh_key\` as \`sk\` left join \`project_ssh_key\` as \`ps\` on \`sk\`.\`id\` = \`ps\`.\`skid\` left join \`customer_ssh_key\` as \`cs\` on \`sk\`.\`id\` = \`cs\`.\`skid\` limit 1"`;

exports[`Sql allowedToModify if not admin, should create statement with IN clauses 1`] = `"select \`sk\`.\`id\`, \`ps\`.\`pid\`, \`cs\`.\`cid\`, IF(COUNT(sk.id) > 0, 1, 0) as allowed from \`ssh_key\` as \`sk\` left join \`project_ssh_key\` as \`ps\` on \`sk\`.\`id\` = \`ps\`.\`skid\` left join \`customer_ssh_key\` as \`cs\` on \`sk\`.\`id\` = \`cs\`.\`skid\` where \`sk\`.\`id\` = 1 and ((\`cs\`.\`cid\` in ('1', '2') or 1 = 0) or (cs.cid IS NULL and ps.pid IS NULL)) limit 1"`;

exports[`Sql selectAllSshKeys if admin, should create statement without IN clause 1`] = `"select \`sk\`.* from \`ssh_key\` as \`sk\` left join \`project_ssh_key\` as \`ps\` on \`sk\`.\`id\` = \`ps\`.\`skid\` left join \`customer_ssh_key\` as \`cs\` on \`sk\`.\`id\` = \`cs\`.\`skid\`"`;

exports[`Sql selectAllSshKeys if not admin, should create statement with IN clauses 1`] = `"select \`sk\`.* from \`ssh_key\` as \`sk\` left join \`project_ssh_key\` as \`ps\` on \`sk\`.\`id\` = \`ps\`.\`skid\` left join \`customer_ssh_key\` as \`cs\` on \`sk\`.\`id\` = \`cs\`.\`skid\` where ((\`cs\`.\`cid\` in ('1', '2') or 1 = 0) or (cs.cid IS NULL and ps.pid IS NULL))"`;

exports[`Sql selectSshKey should create a proper query 1`] = `"select * from \`ssh_key\` where \`id\` = 1"`;

exports[`Sql selectUnassignedSshKeys should create a proper query 1`] = `"select \`sk\`.* from \`ssh_key\` as \`sk\` left join \`project_ssh_key\` as \`ps\` on \`sk\`.\`id\` = \`ps\`.\`skid\` left join \`customer_ssh_key\` as \`cs\` on \`sk\`.\`id\` = \`cs\`.\`skid\` where ps.pid IS NULL and cs.cid IS NULL"`;

exports[`Sql updateSshKey should create a proper query 1`] = `"update \`ssh_key\` set \`keyType\` = 'ssh-rsa' where \`id\` = 1"`;
